downloadertask/IspwDownloader.ts

```diff
@@ -214,41 +218,68 @@ export class IspwDownloader {
-            cliArguments.get("userId"), '-pass', cliArguments.get("password"), '-targetFolder', cliArguments.get("windowsSourceDownloadLocation"),
-            '-scm', 'ISPW', '-ispwServerConfig', cliArguments.get("runtimeConfig"),'-ispwServerStream', cliArguments.get("stream"), '-ispwServerApp', cliArguments.get("application"),
-            '-ispwServerLevel', cliArguments.get("repositoryLevel"), '-ispwDownloadAll', cliArguments.get("downloadUnchangedSource"),
-            '-ispwDownloadIncl', cliArguments.get("downloadIncludes"), '-ispwLevelOption', cliArguments.get("levelOption"),
-            '-ispwComponentType',cliArguments.get("componentTypes"),'-ispwFolderName',cliArguments.get("applicationRootFolderNames"),
-            '-ispwDownloadWithCompileOnly',cliArguments.get("downloadCompileOnly"), '-ispwFilterFiles',
-            cliArguments.get("ispwFilterFiles"),'-ispwFilterFolders',cliArguments.get("ispwFilterFolders")
-        ]);
-        console.log(ls);
-
-        ls.stdout.on('data', function (data) {
-            console.log('stdout: ' + data);
-        });
-
-        ls.stderr.on('data', function (data) {
-            console.log('stderr: ' + data);
-        });
-
-        ls.on('exit', function (code) {
-            console.log('child process exited with code ' + code);
-            if(code!=null && code!=0)
-            {
-                tl.setResult(tl.TaskResult.Failed, " An error may have occurred. Please see task logs or see the log file: "+cliArguments.get("windowsTopazWorkbenchCliHome")+"\\TopazBatchWkspc\\.metadata\\.log.");
-            }
-        });
-
-        
-
+        let command = this.getCommand(cliArguments);
+        console.log(command);
+        //Calling command
+        if (command != undefined) {
+            var ls = spawn(command, ['-host', cliArguments.get("host"), '-port', cliArguments.get("port"), '-code', cliArguments.get("codePage"), '-id',
+                cliArguments.get("userId"), '-pass', cliArguments.get("password"), '-targetFolder', cliArguments.get("sourceDownloadLocation"),
+                '-scm', 'ISPW', '-ispwServerConfig', cliArguments.get("runtimeConfig"), '-ispwServerStream', cliArguments.get("stream"), '-ispwServerApp', cliArguments.get("application"),
+                '-ispwServerSubAppl', cliArguments.get("subAppl"), '-ispwServerLevel', cliArguments.get("repositoryLevel"), '-ispwDownloadAll', cliArguments.get("downloadUnchangedSource"),
+                '-ispwDownloadIncl', cliArguments.get("downloadIncludes"), '-ispwLevelOption', cliArguments.get("levelOption"),
+                '-ispwComponentType', cliArguments.get("componentTypes"), '-ispwFolderName', cliArguments.get("applicationRootFolderNames"),
+                '-ispwDownloadWithCompileOnly', cliArguments.get("downloadCompileOnly"), '-ispwFilterFiles',
+                cliArguments.get("ispwFilterFiles"), '-ispwFilterFolders', cliArguments.get("ispwFilterFolders")
+            ]);
+            console.log(ls);
```

Comment:
is this console log required ?

Reply:
I think this will be helpful for the user to see the whole command and detect if there are any errors